scanner/datastore/postgres/manifest_metadata.go

```diff
@@ -0,0 +1,138 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/quay/claircore"
+	"github.com/quay/zlog"
+)
+
+func (i *indexerMetadataStore) MigrateManifests(ctx context.Context) ([]string, error) {
+	if i.store == nil {
+		return nil, errors.New("indexer store not defined")
+	}
+
+	ctx = zlog.ContextWithValues(ctx, "component", "datastore/postgres/indexerMetadataStore.MigrateManifests")
+
+	// insertMissingManifests inserts missing manifests from the manifest table into manifest_metadata,
+	// and it sets the expiration time to some random time in the future between seven days from now
+	// and thirty days from now.
+	const insertMissingManifests = `
+		INSERT INTO manifest_metadata (manifest_id, expiration)
+		SELECT m.hash, (now() AT TIME ZONE 'utc') + (make_interval(days => 23) * random()) + make_interval(days => 7)
+		FROM manifest m
+		WHERE NOT EXISTS (
+			SELECT FROM manifest_metadata mm WHERE mm.manifest_id = m.hash
+		)
+		RETURNING manifest_id`
+
+	rows, err := i.pool.Query(ctx, insertMissingManifests)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var missingManifests []string
+	for rows.Next() {
+		var manifestID string
+		if err := rows.Scan(&manifestID); err != nil {
+			zlog.Warn(ctx).Err(err).Msg("scanning manifest row")
+			continue
+		}
+		missingManifests = append(missingManifests, manifestID)
+	}
+	if err := rows.Err(); err != nil {
+		zlog.Warn(ctx).Err(err).Msg("reading manifest rows")
+	}
+	if len(missingManifests) > 0 {
+		zlog.Debug(ctx).Strs("migrated_manifests", missingManifests).Msg("migrated missing manifest metadata")
+	}
+	zlog.Info(ctx).Int("migrated_manifests", len(missingManifests)).Msg("migrated missing manifest metadata")
```

Comment:
nit: the same is also logged in the caller `scanner/indexer/manifest/manager.go` - consider removing this log or the other if appropriate to do so.